### ISSUE TITLE
Bad Intrusive Behavior with Rails 3.1

### ISSUE THREAD

**Issue by whoshallsucceed** [mlomnicki/automatic_foreign_key#6]
Hi there!

I come here because I got some surprises. I like the job the gem does. It works quite well. But it works just for migrations. After that, I have to remove this gem from my application. Else nothing works.

If I inspect any model, I got:

```
$ rails c
Loading development environment (Rails 3.1.0.rc4)
ruby-1.9.2-p180 :001 > User
(Object doesn't support #inspect)
 =>  
ruby-1.9.2-p180 :002 >
```

So, of course, after that, any method calls will screw.

I can't even launch my app. I got some message like:

```
Started GET "/" for 127.0.0.1 at 2011-07-21 18:08:27 +0200

ActionController::RoutingError (You have a nil object when you didn't expect it!
You might have expected an instance of Array.
The error occurred while evaluating nil.map):
  app/controllers/users_controller.rb:1:in `<top (required)>'

Rendered .bundle/ruby/1.9.1/gems/actionpack-3.1.0.rc4/lib/action_dispatch/middleware/templates/rescues/routing_error.erb within rescues/layout (0.6ms)
```

Any idea about what I did wrong?

**Comment by mlomnicki** [comment]
@whoshallsucceed thanks for reporting that.

You did everything right. It was an issue of redhillonrails_core dependecy. It's fixed now but prerelease version must be used.

Your Gemfile should included these lines:

```
gem "automatic_foreign_key", "~> 1.3.0"
gem "redhillonrails_core", "~> 2.0.0.pre"
```

As soon as redhillonrails_core 2.0.0 is released you can remove explicit reference to 2.0.0.pre

Please try it and let me know if it works for you.

**Comment by whoshallsucceed** [comment]
Yop! Works fine With redhillonrails_core in prerelease :) Let me know when it's released :)
